### PR TITLE
Better facebook error handling

### LINKED ISSUE #352: If facebook email permission is not granted, no error handling occurs
If a user does not grant the permission for the email address:

image

There is neither an error handling in `FacebookProvider.java` (the caught `JSONException` only closes the loading dialog):

```
GraphRequest request = GraphRequest.newMeRequest(
                loginResult.getAccessToken(),
                new GraphRequest.GraphJSONObjectCallback() {
                    @Override
                    public void onCompleted(JSONObject object, GraphResponse response) {
                        try {
                            String email = object.getString("email");
                            mCallbackObject.onSuccess(createIDPResponse(loginResult, email));
                        } catch (JSONException e) {
                            e.printStackTrace();
                            mCallbackObject.onFailure(new Bundle());
                        }
                    }
                });
```

nor an error handling in `AuthMethodPickerActivity.java`

```
    @Override
    public void onFailure(Bundle extra) {
        // stay on this screen
        mActivityHelper.dismissDialog();
    }
```

According to the facebook permission handling guidlines https://developers.facebook.com/docs/facebook-login/handling-declined-permissions#reprompt a reprompt should be shown. This reprompt should explain why the email permission is needed

### PR DESCRIPTION
**Pull request by amandle**
Resolves #352 and #263. Provides better error handling for Facebook Graph requests.

### CODE REVIEW

File: auth/src/main/java/com/firebase/ui/auth/provider/FacebookProvider.java [hunk 1]
```diff
@@ -125,12 +126,24 @@ public void onSuccess(final LoginResult loginResult) {
...
+                        }
+                        if (object == null) {
+                            Log.w(TAG, "Received null response from Facebook GraphRequest");
+                            mCallbackObject.onFailure(new Bundle());
+                        } else {
+                            try {
+                                String email = object.getString("email");
+                                mCallbackObject.onSuccess(createIDPResponse(loginResult, email));
+                            } catch (JSONException e) {
+                                e.printStackTrace();
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/provider/FacebookProvider.java]: Log.e(...) instead of e.printStackTrace().

**Comment by samtstern** [comment]
One comment, LGTM to merge once that's fixed.

**Comment by SUPERCILEX** [comment]
That DDoS attack was nuts! At least GitHub is back now...

Travis is going to take a while to catch up on all the backlog though.

**Comment by amandle** [comment]
Looks like Travis is failing because of github being down intermittently I will try again later

**Comment by samtstern** [comment]
Travis passed, so merge at will.

**Pull request merged**